### PR TITLE
add domain layer module based on clean architecture

### PR DESCRIPTION
**Pull request by pariami**
add domain layer module

### CODE REVIEW

**Review by viamgr — APPROVED** [review]
Nice job Paria

File: domain/build.gradle.kts [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    kotlinOptions {
+        jvmTarget = "1.8"
+    }
+}
+
+dependencies {
+
+    implementation("androidx.core:core-ktx:1.9.0")
+    implementation("androidx.appcompat:appcompat:1.6.1")
+    implementation("com.google.android.material:material:1.9.0")
```
Comment (ms-dehghani) [inline domain/build.gradle.kts:39]: Do we need this dependency?

File: domain/build.gradle.kts [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    }
+    kotlinOptions {
+        jvmTarget = "1.8"
+    }
+}
+
+dependencies {
+
+    implementation("androidx.core:core-ktx:1.9.0")
+    implementation("androidx.appcompat:appcompat:1.6.1")
```
Comment (ms-dehghani) [inline domain/build.gradle.kts:38]: Do we need this dependency?

File: domain/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        targetCompatibility = JavaVersion.VERSION_1_8
+    }
+    kotlinOptions {
+        jvmTarget = "1.8"
+    }
+}
+
+dependencies {
+
+    implementation("androidx.core:core-ktx:1.9.0")
```
Comment (ms-dehghani) [inline domain/build.gradle.kts:37]: Do we need this dependency?

File: domain/build.gradle.kts [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    implementation("androidx.appcompat:appcompat:1.6.1")
+    implementation("com.google.android.material:material:1.9.0")
+    testImplementation("junit:junit:4.13.2")
+    androidTestImplementation("androidx.test.ext:junit:1.1.5")
+    androidTestImplementation("androidx.test.espresso:espresso-core:3.5.1")
```
Comment (ms-dehghani) [inline domain/build.gradle.kts:42]: Do we need this dependency?

File: domain/src/androidTest/java/com/trainning/app/domain/ExampleInstrumentedTest.kt
Comment (ms-dehghani) [inline domain/src/androidTest/java/com/trainning/app/domain/ExampleInstrumentedTest.kt:1]: Please remove sample test case

File: domain/src/test/java/com/trainning/app/domain/ExampleUnitTest.kt [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+ * Example local unit test, which will execute on the development machine (host).
+ *
+ * See [testing documentation](http://d.android.com/tools/testing).
+ */
+class ExampleUnitTest {
+    @Test
+    fun addition_isCorrect() {
+        assertEquals(4, 2 + 2)
+    }
+}
```
Comment (ms-dehghani) [inline domain/src/test/java/com/trainning/app/domain/ExampleUnitTest.kt:17]: Please remove sample test case

**Review by viamgr — APPROVED** [review]

**Review by ms-dehghani — APPROVED** [review]

**Pull request merged**